pkg/manifests/manifests.go

```diff
@@ -1659,6 +1662,36 @@ func (f *Factory) PrometheusK8s(grpcTLS *v1.Secret, trustedCABundleCM *v1.Config
 	return p, nil
 }
 
+func (f *Factory) setupQueryLogFile(p *monv1.Prometheus) error {
+	queryLogFile := f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.QueryLogFile
+	dirPath := filepath.Dir(queryLogFile)
+	if dirPath == "/" {
+		return errors.Wrap(ErrConfigValidation, `query log file can't be stored on the root directory`)
+	}
+
+	p.Spec.QueryLogFile = queryLogFile
+	if dirPath == "/prometheus" || dirPath == "/dev" {
+		return nil
+	}
+
+	p.Spec.Volumes = append(
+		p.Spec.Volumes,
+		v1.Volume{
+			Name: "query-log-file",
+			VolumeSource: v1.VolumeSource{
+				EmptyDir: p.Spec.Storage.EmptyDir,
```

Comment:
we want a default emptyDir here:

```suggestion
				EmptyDir: &v1.EmptyDirVolumeSource{},
```